docs/root/intro/arch_overview/upstream/dns_resolution.rst

```diff
@@ -0,0 +1,24 @@
+.. _arch_overview_dns_resolution:
+
+DNS Resolution
+==============
+
+Many Envoy components resolve DNS: different cluster types (
+:ref:`strict dns <arch_overview_service_discovery_types_strict_dns>`,
+:ref:`logical dns <arch_overview_service_discovery_types_logical_dns>`);
+the :ref:`dynamic forward proxy <arch_overview_http_dynamic_forward_proxy>` system (which is
+composed of a cluster and a filter);
+the udp :ref:`dns filter <arch_overview_dns_filter>`, etc.
+Envoy uses `c-ares <https://github.com/c-ares/c-ares>`_ as a third party DNS resolution library.
+On Apple OSes Envoy additionally offers resolution using Apple specific APIs via the
+``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime feature.
```

Comment:
When we clean up the runtime feature flag for this (which IMO we can do soon if no one complains, which they won't), please remember to clean this up also.

Reply:
For sure, will search and update all occurrences.